Multi-app workflows: improve docs and images (#4878)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-multi-app.md

- Implementation of a workflow spans different programming languages based on team expertise or existing codebases.
- Different team boundaries or microservice ownership.

<img src="/images/workflow-overview/workflow-multi-app-complex.png" width=800 alt="Diagram showing multi-application complex workflow">

> The image depicts a complex, multi-language workflow scenario with the following pattern:

Reviewer: ```suggestion
The diagram below shows an example scenario of a complex workflow that orchestrates across multiple applications that are written in different languages. Each applications' main steps and activities are;
```

• **App1: Main Workflow Service** - Top-level orchestrator that coordinates the entire ML pipeline
- Starts the process
- Calls data processing activities on App2
- Calls ML training child workflow on App3
- Calls model deployment on App4
- Ends the complete workflow
- **Language: Java**

• **App2: Data Processing Pipeline** - **GPU activities** only
- Data Ingesting Activity (GPU-accelerated)
- Feature Engineering Activity (GPU-accelerated)
- Returns completion signal to Main Workflow
- **Language: Go**

• **App3: ML Training Child Workflow** - Contains a child workflow and activities
- Child workflow orchestrates:
  - Data Processing Activity
  - Model Training Activity (GPU-intensive)
  - Model Validation Activity
- Triggered by App2's activities completing
- Returns completion signal to Main Workflow
- **Language: Java**

• **App4: Model Serving Service** - **Beefy GPU app** with activities only
- Model Loading Activity (GPU memory intensive)
- Inference Setup Activity (GPU-accelerated inference)
- Triggered by App3's workflow completing
- Returns completion signal to Main Workflow
- **Language: Go**

## Multi-application workflows

> Like all building blocks in Dapr, workflow execution routing is based on the [App ID of the hosting Dapr application]({{% ref "security-concept.md#application-identity" %}}).

Reviewer: ```suggestion
Workflow execution routing is based on the [App ID of the hosting Dapr application]({{% ref "security-concept.md#application-identity" %}}).
```

By default, the full workflow execution is hosted on the app ID that started the workflow. This workflow can be executed across any replicas of that app ID, not just the single replica which scheduled the workflow.

It is possible to execute activities and child workflows on different app IDs by specifying the target app ID parameter, inside the workflow execution code.

> Upon execution, the target app ID will execute the activity or child workflow, and return the result to the parent workflow of the originating app ID.

Reviewer: ```suggestion
Upon execution, the target app ID executes the activity or child workflow, and returns the result to the parent workflow of the originating app ID.
```

> Like other building blocks and resources in Dapr, workflows are scoped to a single namespace.

Reviewer: ```suggestion
Like other API building blocks and resources in Dapr, workflows are scoped to a single namespace.
```

This means that all app IDs involved in a multi-application workflow must be in the same namespace.

> Similarly, all app IDs must use the same actor state store.

Reviewer: ```suggestion
Similarly, all app IDs must use the same workflow (or actor) state store.
```

> Finally, the target app ID must have the activity or child workflow defined and registered, otherwise the parent workflow will retry indefinitely.

Reviewer: ```suggestion
Finally, the target app ID must have the activity or child workflow defined and registered, otherwise the parent workflow retries indefinitely.
```

{{% /alert %}}

{{% alert title="Important Limitations" color="warning" %}}
**SDKs supporting multi-application workflows** - Multi-application workflows are used via the SDKs. 

> Currently the following is supported:

Reviewer: ```suggestion
Currently the following are supported:
```
